glider_qc/glider_qc.py

```diff
@@ -338,53 +338,53 @@ def create_location_flag_variable(self, ndim, flag):
         ncvar.references = 'The GDAC uses a modified version of the location test described in https://gliders.ioos.us/files/Manual-for-QC-of-Glider-Data_05_09_16.pdf'
         ncvar.qartod_package = 'The GDAC location test does not use the algorithm from https://github.com/ioos/ioos_qc/blob/main/ioos_qc/qartod.py (location_test) but instead relies on the statistical median of the lat/lon arrays'
         ncvar.dac_comment = 'The FAIL flag is applied if the profile_(lat,lon) value exceeds 3 standard deviations above the mean of the average lat/lon arrays'
-        ncvar.ioos_category = 'Quality' 
-        
+        ncvar.ioos_category = 'Quality'
+
         return ncvar
 
     def check_location(self, ncf, report):
         '''
         Check the glider track lon and lat coordinates for outliers.
         If an outlier is detected:
             - Copy the profile_lat/lon variables onto new variables to preserve the original data.
-            - Replace the profile_lat/lon data with the median of the lat or lon arrays.  
-        :param ncf: netCDF4._netCDF4.Dataset 
+            - Replace the profile_lat/lon data with the median of the lat or lon arrays.
+        :param ncf: netCDF4._netCDF4.Dataset
         :param report: string reporting on issues
         '''
-        profile_lat = ncf.variables['profile_lat']
-        profile_lon = ncf.variables['profile_lon']
+        profile_lat = ncf.variables['profile_lat'][0]
+        profile_lon = ncf.variables['profile_lon'][0]
 
         lat = ncf.variables['lat']
         lon = ncf.variables['lon']
 
-        if  ~np.isnan(profile_lat[:]) \
-            or ~np.isnan(profile_lon[:]):
+        if  ~(np.isnan(profile_lat) & np.ma.is_masked(profile_lat) &
```

Comment:
`np.isnan` was failing when trying to process whole variable context rather than single scalar.  Mask also needs to be considered as some things may be non-NaN but still masked out by `_FillValue`, for example.